### PR TITLE
feat(ts-type-expand-plugin): support `vue` language

### PR DESCRIPTION
**Pull request by zcf0508**
1. install `vue.volar` extension above 2.0.16
2. enable the `Hybrid Mode`

image

---

image

### CODE REVIEW

**Comment by zcf0508** [comment]
https://github.com/mxsdev/ts-type-explorer/pull/45

**Comment by d-kimuson** [comment]
Related https://github.com/d-kimuson/ts-type-expand/issues/330

**Comment by d-kimuson** [comment]
Thank you for your contribution!

I've been wanting to support Vue but haven't had the chance to start, so I really appreciate this. Everything seems to be working fine on my end. I'll review the changes, and if there are no issues, I'll merge them as they are.

File: packages/ts-type-expand-plugin/src/server/context.ts [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+export const { setTypescript, getTypescript } = (() => {
+  let ts: __ts | undefined = undefined
+
+  return {
+    setTypescript: (_ts: __ts) => {
+      ts = _ts
+    },
+    getTypescript: (): __ts | undefined => ts,
+  }
+})()
```
Comment (d-kimuson) [inline packages/ts-type-expand-plugin/src/server/context.ts:21]: nits:

```suggestion
export const { setTypeScript, getTypeScript } = (() => {
  let ts: __ts | undefined = undefined

  return {
    setTypeScript: (_ts: __ts) => {
      ts = _ts
    },
    getTypeScript: (): __ts | undefined => ts,
  }
})()
```

File: packages/ts-type-expand-plugin/src/server/context.ts [hunk 2]
```diff
@@ -19,10 +33,12 @@ export const { setCreateInfo, getCreateInfo } = (() => {
 
 export const createContext = async (): Promise<Context> => {
   const createInfo = getCreateInfo()
+  const ts = getTypescript()
```
Comment (d-kimuson) [inline packages/ts-type-expand-plugin/src/server/context.ts:36]: nits

```suggestion
  const ts = getTypeScript()
```

File: packages/ts-type-expand-plugin/src/service/vue.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          )
+        if (serviceScript) {
+          const map = language.maps.get(serviceScript.code, vFile.id)
+          if (map) {
+            for (const [generatedOffset, mapping] of toGeneratedOffsets(
+              vFile,
+              map,
+              startPos,
+            )) {
+              console.log(JSON.stringify({ generatedOffset, mapping }))
```
Comment (d-kimuson) [inline packages/ts-type-expand-plugin/src/service/vue.ts:116]: want: Please use project defined logger for logging.

ref: https://github.com/d-kimuson/ts-type-expand/blob/76443e30e14d675b34947bbca1dd1fa58f64a106/packages/ts-type-expand-plugin/src/index.ts#L17-L20

File: packages/ts-type-expand-plugin/src/index.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import type { server } from 'typescript/lib/tsserverlibrary.js'
 import { logger } from './logger.js'
 import { pluginConfigurationSchema } from './schema.js'
 import { registerApp } from './server/app.js'
-import { setCreateInfo } from './server/context.js'
+import { setCreateInfo, setTypescript } from './server/context.js'
```
Comment (d-kimuson) [inline packages/ts-type-expand-plugin/src/index.ts:8]: nits

```suggestion
import { setCreateInfo, setTypeScript } from './server/context.js'
```

File: packages/ts-type-expand-plugin/src/index.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ const factory: server.PluginModuleFactory = (_mod) => {
       })
 
       setCreateInfo(info)
+      setTypescript(_mod.typescript)
```
Comment (d-kimuson) [inline packages/ts-type-expand-plugin/src/index.ts:23]: nits

```suggestion
      setTypeScript(_mod.typescript)
```

**Review by d-kimuson — CHANGES_REQUESTED** [review]
Overall, it looks good. I’ve made a few minor comments, so please address those.

**Review by d-kimuson — APPROVED** [review]
Thanks so much!
I’ll let you know once it's released.

**Comment by d-kimuson** [comment]
I’ve released a new version with added support for Vue!
Release: https://github.com/d-kimuson/ts-type-expand/releases/tag/v1.2.0

**Pull request merged**